src/payment/strategies/klarna/klarna-payment-strategy.ts

```diff
@@ -96,18 +96,49 @@ export default class KlarnaPaymentStrategy implements PaymentStrategy {
         return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
             .then(state => new Promise<KlarnaLoadResponse>((resolve, reject) => {
                 const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+                const shippingAddress = state.shippingAddress.getShippingAddress();
+                const billingAddress = state.billingAddress.getBillingAddress();
 
                 if (!paymentMethod) {
                     throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
                 }
 
+                if (!billingAddress) {
+                    throw new MissingDataError(MissingDataErrorType.MissingBillingAddress);
+                }
+
+                if (!shippingAddress) {
```

Comment:
Shipping address is optional. For example, Digital Products don´t require a shipping address.